packages/api/src/clickhouse/index.ts

```diff
@@ -904,6 +912,588 @@ export const getMetricsChart = async ({
+  );
+
+  const rows = await client.query({
+    query,
+    format: 'JSON',
+    clickhouse_settings: {
+      additional_table_filters: buildLogStreamAdditionalFilters(
+        tableVersion,
+        teamId,
+      ),
+    },
+  });
+
+  const result = await rows.json<
+    ResponseJSON<{
+      ts_bucket: number;
+      group: string;
+      [series_data: `series_${number}.data`]: number;
+    }>
+  >();
+  return result;
+};
+
+export const getMultiSeriesChart = async ({
+  series,
+  endTime,
+  granularity,
+  maxNumGroups,
+  propertyTypeMappingsModel,
+  startTime,
+  tableVersion,
+  teamId,
+  seriesReturnType = SeriesReturnType.Column,
+}: {
+  series: z.infer<typeof chartSeriesSchema>[];
+  endTime: number; // unix in ms,
+  startTime: number; // unix in ms
+  granularity: string | undefined; // can be undefined in the number chart
+  maxNumGroups: number;
+  propertyTypeMappingsModel?: LogsPropertyTypeMappingsModel;
+  tableVersion: number | undefined;
+  teamId: string;
+  seriesReturnType?: SeriesReturnType;
+}) => {
+  let queries: { query: string; hasGroupBy: boolean }[] = [];
+  if ('table' in series[0] && series[0].table === 'logs') {
+    if (propertyTypeMappingsModel == null) {
+      throw new Error('propertyTypeMappingsModel is required for logs chart');
+    }
+
+    queries = await Promise.all(
+      series.map(s => {
+        if (s.type != 'time' && s.type != 'table') {
+          throw new Error(`Unsupported series type: ${s.type}`);
+        }
+
+        return buildEventSeriesQuery({
+          aggFn: s.aggFn,
+          endTime,
+          field: s.field,
+          granularity,
+          groupBy: s.groupBy[0],
+          maxNumGroups,
+          propertyTypeMappingsModel,
+          q: s.where,
+          sortOrder: s.type === 'table' ? s.sortOrder : undefined,
+          startTime,
+          tableVersion,
+          teamId,
+        });
+      }),
+    );
+  } else if ('table' in series[0] && series[0].table === 'metrics') {
+    queries = await Promise.all(
+      series.map(s => {
+        if (s.type != 'time' && s.type != 'table') {
+          throw new Error(`Unsupported series type: ${s.type}`);
+        }
+        if (s.field == null) {
+          throw new Error('Metric name is required');
+        }
+        if (s.metricDataType == null) {
+          throw new Error('Metric data type is required');
+        }
+
+        return buildMetricSeriesQuery({
+          aggFn: s.aggFn,
+          endTime,
+          name: s.field,
+          granularity,
+          groupBy: s.groupBy[0],
+          // maxNumGroups,
+          q: s.where,
+          // sortOrder: s.sortOrder,
+          startTime,
+          teamId,
+          dataType: s.metricDataType,
+        });
+      }),
+    );
+  }
+
+  return queryMultiSeriesChart({
+    maxNumGroups,
+    tableVersion,
+    teamId,
+    seriesReturnType,
+    queries,
+  });
+};
+
+export const getMultiSeriesChartLegacyFormat = async ({
```

Comment:
I think we need to test this guy to make sure it outputs the same data as the original chart method